protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -893,7 +917,8 @@ impl PoolChannelFactory {
         &mut self,
         m: &SetNewPrevHashFromTp<'static>,
     ) -> Result<u32, Error> {
-        let job_id = self.job_creator.on_new_prev_hash(m).unwrap_or(0);
+        // since we have staged phash 0 is no longer reserved and 1 is the default
```

Comment:
What do you mean?

Reply:
I noticed the first SNPH getting sent out was with job_id=0 whereas the first NEMJ getting sent was with job_id=1 so the tproxy was having issues (the first job getting sent to miners was with job_id=0 and we were checking shares against job_id=1. iirc, the `StagedPrevHash` was added so we dont have to have 0 reserved anymore for when a prev hash arrives and there is no template.

Reply:
it may be better to get the JN in Tproxy PR merged first so I can rebase and confirm